pkg/fleet/installer/service/apm_sockets.go

```diff
@@ -0,0 +1,190 @@
+func addEnvsIfNotSet(envs map[string]string, envFile []byte) ([]byte, error) {
+	var buffer bytes.Buffer
+	buffer.Write(envFile)
+
+	if len(envFile) > 0 && envFile[len(envFile)-1] != '\n' {
+		buffer.WriteByte('\n')
+	}
+	for key, value := range envs {
+		// There's a slight gap where there exists an env var in the form of {prefix}{key} but not {key},
+		// but as our env vars all start with DD_ prefixing them won't cause any issue.
+		if !bytes.Contains(envFile, []byte(key+"=")) {
+			buffer.WriteString(fmt.Sprintf("%s=%s\n", key, value))
+		}
+	}
+	return buffer.Bytes(), nil
+}
+
+// verifySocketEnvs verifies that socket environment variables can be sourced
+func verifySocketEnvs(path string) error {
+	// Verify DD_USE_DOGSTATSD
+	cmd := exec.Command("/bin/sh", "-c", fmt.Sprintf(". %s && echo $DD_USE_DOGSTATSD", path))
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("error verifying socket environment variables: %w (%s)", err, output)
+	}
+	output = bytes.TrimSpace(output)
+	if string(output) != "true" {
+		return fmt.Errorf("DD_USE_DOGSTATSD environment variables not set correctly")
+	}
+
+	// Verify DD_APM_RECEIVER_SOCKET
+	cmd = exec.Command("/bin/sh", "-c", fmt.Sprintf(". %s && echo $DD_APM_RECEIVER_SOCKET", path))
+	output, err = cmd.CombinedOutput()
+	output = bytes.TrimSpace(output)
+	if err != nil {
+		return fmt.Errorf("error verifying socket environment variables: %w (%s)", err, output)
+	}
+	if string(output) == "" {
+		return fmt.Errorf("DD_APM_RECEIVER_SOCKET environment variables not set correctly")
+	}
+
+	// Verify DD_DOGSTATSD_SOCKET
+	cmd = exec.Command("/bin/sh", "-c", fmt.Sprintf(". %s && echo $DD_DOGSTATSD_SOCKET", path))
+	output, err = cmd.CombinedOutput()
+	output = bytes.TrimSpace(output)
+	if err != nil {
+		return fmt.Errorf("error verifying socket environment variables: %w (%s)", err, output)
+	}
+	if string(output) == "" {
+		return fmt.Errorf("DD_DOGSTATSD_SOCKET environment variables not set correctly")
+	}
+	return nil
+}
+
+// addSystemDEnvOverrides adds /etc/environment variables to the defined systemd units
+// The unit should contain the .service suffix (e.g. datadog-agent-exp.service)
+//
+// Reloading systemd & restarting the unit has to be done separately by the caller
+func addSystemDEnvOverrides(unit string) error {
+	// Verify that the unit is a DD one
+	if !strings.HasPrefix(unit, "datadog-") {
+		return fmt.Errorf("unit %s is not a Datadog unit", unit)
+	}
+	if !strings.HasSuffix(unit, ".service") {
+		return fmt.Errorf("unit %s is not a service unit", unit)
+	}
+
+	content := []byte("[Service]\nEnvironmentFile=/etc/environment\n")
+
+	// We don't need a file mutator here as we're fully hard coding the content.
+	// We don't really need to remove the file either as it'll just be ignored once the
+	// unit is removed.
+	path := fmt.Sprintf("/etc/systemd/system/%s.d/environment_override.conf", unit)
+	err := os.MkdirAll(filepath.Dir(path), 0755)
```

Comment:
- Why `/etc` instead of `/lib` where we have our stuff usually ?
- A `Mkdir` might be slightly better or are we expecting some of the parent dirs to be missing ?

Reply:
Done in [8072a8a](https://github.com/DataDog/datadog-agent/pull/25577/commits/8072a8ad9228506a7a7628419e6c2891d6dfdd01)